mms/model_service_worker.py

```diff
@@ -10,70 +10,57 @@
 
 """
 ModelServiceWorker is the worker that is started by the MMS front-end.
-Communication message format: JSON message
+Communication message format: binary encoding
 """
 
 # pylint: disable=redefined-builtin
 
+import logging
 import os
 import socket
+import sys
 
 from mms.arg_parser import ArgParser
-from mms.log import log_msg, log_error
 from mms.model_loader import ModelLoaderFactory
-from mms.mxnet_model_service_error import MMSError
-from mms.protocol.otf_message_handler import OtfCodecHandler
-from mms.utils.model_server_error_codes import ModelServerErrorCodes as Err
-from mms.utils.validators.validate_messages import ModelWorkerMessageValidators
+from mms.protocol.otf_message_handler import retrieve_msg, create_load_model_response
 
 MAX_FAILURE_THRESHOLD = 5
 SOCKET_ACCEPT_TIMEOUT = 30.0
-debug = False
-BENCHMARK = False
+DEBUG = False
 
 
 class MXNetModelServiceWorker(object):
     """
     Backend worker to handle Model Server's python service code
     """
     def __init__(self, s_type=None, s_name=None, host_addr=None, port_num=None):
-        if os.environ.get('OMP_NUM_THREADS') is None:
-            os.environ['OMP_NUM_THREADS'] = "1"
+        if os.environ.get("OMP_NUM_THREADS") is None:
+            os.environ["OMP_NUM_THREADS"] = "1"
         self.sock_type = s_type
-        if s_type == 'unix':
+        if s_type == "unix":
             if s_name is None:
-                raise MMSError(Err.INVALID_ARGUMENTS, "Wrong arguments passed. No socket name given.")
+                raise ValueError("Wrong arguments passed. No socket name given.")
             self.sock_name, self.port = s_name, -1
             try:
                 os.unlink(s_name)
             except OSError:
                 if os.path.exists(s_name):
-                    raise MMSError(Err.SOCKET_ERROR, "socket already in use: {}.".format(s_name))
-        elif s_type == 'tcp':
+                    raise RuntimeError("socket already in use: {}.".format(s_name))
+
+        elif s_type == "tcp":
             self.sock_name = host_addr if host_addr is not None else "127.0.0.1"
             if port_num is None:
-                raise MMSError(Err.INVALID_ARGUMENTS, "Wrong arguments passed. No socket port given.")
+                raise ValueError("Wrong arguments passed. No socket port given.")
             self.port = port_num
         else:
             raise ValueError("Incomplete data provided")
 
-        self.model_services = {}
-        self.send_failures = 0
-        self.codec = OtfCodecHandler()
-
-        try:
-            msg = "Listening on port: {}\n".format(s_name)
-            log_msg(msg)
-            socket_family = socket.AF_INET if s_type == 'tcp' else socket.AF_UNIX
-            self.sock = socket.socket(socket_family, socket.SOCK_STREAM)
-
-        except (IOError, OSError) as e:
```

Comment:
Are we sure this exception need not be handled?

Reply:
If these exception is raised, the process will die. What do we need to handle?